diff -Nru a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2018-01-13 01:23:52.000000000 +0100
+++ b/CMakeLists.txt	2018-05-12 10:35:51.559428563 +0200
@@ -287,7 +287,8 @@
 add_subdirectory(COLLADAValidator)
 
 # DAE validator app
-add_subdirectory(DAEValidator)
+# don't build DAE validator app
+#add_subdirectory(DAEValidator)
 
 # Library export
 install(EXPORT LibraryExport DESTINATION ${OPENCOLLADA_INST_CMAKECONFIG} FILE OpenCOLLADATargets.cmake)
